app/modules/ads.js

```diff
@@ -349,6 +365,64 @@ class Ads {
 		}
 	}
 
+	isTopLeaderboardApplicable() {
+		const hasFeaturedVideo = this.getTargetingValue('hasFeaturedVideo'),
+			isHome = this.getTargetingValue('pageType') === 'home',
+			$pi = $('.portable-infobox'),
+			$pageHeader = $('.wiki-page-header');
+
+		return isHome || $pi.length > 0 || ($pageHeader.length > 0 && !hasFeaturedVideo);
```

Comment:
nitpick: what about:
```
portableInfoboxPresent = $('.portable-infobox').length > 0,
pageHeaderPresent = $('.wiki-page-header').length > 0
```
and then in final check it would be:
```
isHome || portableInfoboxPresent || (pageHeaderPresent && !hasFeaturedVideo)
```
??